core/trino-spi/src/main/java/io/trino/spi/block/Int128ArrayBlock.java

```diff
@@ -155,8 +155,11 @@ public boolean hasNull()
     @Override
     public boolean isNull(int position)
     {
+        if (valueIsNull == null) {
```

Comment:
What's the motivation for lifting this up as its own condition here? Seems like the previous implementation handled this already, and the `checkReadablePosition` call should arguably happen with or without the `valueIsNull` array being present to catch incorrect logic in more scenarios.

Reply:
@pettyjamesm please see the contract for `io.trino.spi.block.Block#isNull`
```
/**
     * Is the specified position null?
     *
     * @throws IllegalArgumentException if this position is not valid. The method may return false
     * without throwing exception when there are no nulls in the block, even if the position is invalid
     */
```
We made similar change in https://github.com/trinodb/trino/pull/10264

Reply:
I'm not opposed to making this change, but might point out that we _could_ consider making this assertions so that we don't have the runtime performance hit in production while still getting early detection of logical issues in tests. I know that's against the style guide we've used historically, but maybe worth reconsidering inside of these access paths that we use in performance sensitive loops.